go/vt/tabletserver/grpcqueryservice/server.go

```diff
@@ -0,0 +1,190 @@
+// Copyright 2015, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package grpcqueryservice
+
+import (
+	mproto "github.com/youtube/vitess/go/mysql/proto"
+	"github.com/youtube/vitess/go/vt/callinfo"
+	"github.com/youtube/vitess/go/vt/servenv"
+	"github.com/youtube/vitess/go/vt/tabletserver"
+	"github.com/youtube/vitess/go/vt/tabletserver/proto"
+	"github.com/youtube/vitess/go/vt/tabletserver/queryservice"
+	"golang.org/x/net/context"
+
+	pb "github.com/youtube/vitess/go/vt/proto/query"
+	pbs "github.com/youtube/vitess/go/vt/proto/queryservice"
+)
+
+// Query is the gRPC query service implementation.
+// It implements the queryservice.QueryServer interface.
+type Query struct {
+	server queryservice.QueryService
+}
+
+// New returns a new server. It is public for unit tests to use.
+func New(server queryservice.QueryService) *Query {
+	return &Query{server}
+}
+
+// GetSessionId is part of the queryservice.QueryServer interface
+func (q *Query) GetSessionId(ctx context.Context, request *pb.GetSessionIdRequest) (response *pb.GetSessionIdResponse, err error) {
+	defer q.server.HandlePanic(&err)
+
+	sessionInfo := new(proto.SessionInfo)
+	gsiErr := q.server.GetSessionId(&proto.SessionParams{
+		Keyspace: request.Keyspace,
```

Comment:
We just talked about this, two things for the record:
- they're not pointers any more, just regular values.
- the panic handling should catch the most horrendous problems where the query is not what we expect at all.
So I think all in all we're good.